storage/core/src/main/java/com/hortonworks/streamline/storage/Storable.java

```diff
@@ -85,4 +85,11 @@
      */
     void setId(Long id);
 
+    /**
+     * Returns if the storable should be cached or not
+     */
+    default boolean isCacheable() {
```

Comment:
FileBlob contains an `InputStream` which is basically invalid after the content is sent out, so should not be cached. Other cases could be things like large `byte[]` which we dont want to be cached.